docs/book/src/developer/providers/ipam.md

```diff
@@ -0,0 +1,81 @@
+# IPAM Provider Specification
+
+## Overview
+
+The IPAM provider is responsible for handling the IP addresses for the machines in a cluster.
+
+IPAM providers are optional when using Cluster API. Infrastructure providers need to [implement explicit support](#infrastructure-provider) to be usable in conjunction with IPAM providers.
+
+<aside class="note">
+
+Note that the IPAM contract is single-stack. If you need both v4 and v6 addresses, two pools and two IPAddressClaims are necessary.
+
+</aside>
+
+## Data Types
+
+An IPAM provider must define one or more API types for IP address pools. The types:
+
+`1. Must belong to an API group served by the Kubernetes apiserver 
+2. Must be implemented as a CustomResourceDefinition. 
+   The CRD name must have the format produced by `sigs.k8s.io/cluster-api/util/contract.CalculateCRDName(Group, Kind)`. 
+3. Must have the standard Kubernetes "type metadata" and "object metadata"
+`4. Should have a status.conditions field with the following:
+   1. A Ready condition to represent the overall operational state of the component. It can be based on the summary of more detailed conditions existing on the same object, e.g. instanceReady, SecurityGroupsReady conditions.
```

Comment:
Wondering how this works in the context of https://github.com/kubernetes-sigs/cluster-api/pull/10897 (the migration to metav1.Condition)

Reply:
I'm updating contracts to take into account changes introduced by #10897, I will take care of this one too after this PR merge